### ISSUE TITLE
Keract for non convolutional models

### ISSUE THREAD

**Issue by Lecia321** [philipperemy/keract#116]
Hi there,
Is there possibly an example/tutorial of how to use keract when you have a model that does not process images. As example a simple LSTM with soft max function.

I would like to see how one could use the functions keract offers in that way. Will be very useful to help explain predictions.

**Comment by philipperemy** [comment]
@Lecia321 well keract gives you the outputs of each of your layers. So you can adapt it to your problem. At the moment, there's no example dealing with visualisation for LSTM. But there's a very good tutorial about it (for text generation): 

- http://karpathy.github.io/2015/05/21/rnn-effectiveness/
- http://blog.echen.me/
- https://stats.stackexchange.com/questions/323782/how-to-visualize-char-rnn-model

**Comment by Lecia321** [comment]
Thank you @philipperemy for the reference to the tutorials. They will definitely help since I am very new to keract.

**Comment by philipperemy** [comment]
@Lecia321 welcome! That would be great if you can share some of your code after that others can re-use for visualizing non conv layers. Good luck!

**Comment by Yannik1337** [comment]
Kinda related to _visualizing_ layers:

Extract a layers output, then write it to file. Load the file into TensorBoard (or projector.tensorflow.org), then visualize it using tSNE or PCA.

Sample code, x is a batch to be profiled, y are the according labels.


```
import io

class LayerOutputCallback(keras.callbacks.Callback):
    __slots__=["x", "y", "layer_name", "counter", "frequency"]
    def __init__(self, x, y, layer_name, frequency):
      super(LayerOutputCallback, self).__init__()
      self.x = x
      self.y = y
      print(self.y.shape)
      self.layer_name = layer_name
      self.counter = 0
      self.frequency = frequency   

    def on_epoch_end(self, epoch, logs=None):
      self.counter += 1
      if self.counter == self.frequency: #calculate intermediate ouput every frequency epochs
        self.counter = 0 #reset the counter
        self.get_layer_output()
    
    def get_layer_output(self):
      intermediate_model = tf.keras.Model(inputs = self.model.inputs, outputs=self.model.get_layer(name=self.layer_name).output)
      layer_output = intermediate_model(self.x)
      print("\n Now outputing layer features")
      #print(layer_output.shape)
      

      #for i in range(5):
      #  print(self.y[i], layer_output[i])

      out_v = io.open('vectors.tsv', 'a', encoding='utf-8')
      out_m = io.open('metadata.tsv', 'a', encoding='utf-8')

      for index in range(len(self.y)):
        vec = layer_output[index] 
        out_v.write('\t'.join([str(x) for x in vec.numpy()]) + "\n")
        out_m.write(str(self.y[index]) + "\n")
      out_v.close()
      out_m.close()
      print("\n")
```
